components/builder-api/src/server/resources/origins.rs

```diff
@@ -516,46 +516,61 @@ fn upload_origin_key(req: HttpRequest,
                      -> HttpResponse {
     let (origin, revision) = path.into_inner();
 
```

Comment:
I might add a comment to the effect:

We return conflict when uploading a pre-existing key even if the user doesn't have permissions to actually upload the key. This is a backward compatibility workaround needed when RBAC was introduced. `hab pkg upload` optimistically uploads keys as well as packages, but the RBAC changes only give maintainers upload permissions for packages, not keys. Returning permission denied broke this flow. Somewhat silently accepting the upload was chosen as a way to not break that path while still maintaining the chosen RBAC semantics.